src/language/typescript/common/bundled/utils.ts

```diff
@@ -5,31 +5,16 @@ import { fromRef } from '../../../../utils/fs';
 
 export const utilsRef = fromString('#/utils/utils');
 
-const utils = `
+export const utils = `
 	import { either } from 'fp-ts/lib/Either';
 	import { Type, failure, success, string as tstring } from 'io-ts';
 
-	const getISOTimezoneOffsetString = (offsetMinutes: number): string => {
-		if (offsetMinutes === 0) {
-			return 'Z';
-		}
-
-		const absoluteOffsetMinutes = Math.abs(offsetMinutes);
-		const offsetHours = absoluteOffsetMinutes / 60;
-		const offsetRestMinutes = absoluteOffsetMinutes % 60;
-
-		return \`\${offsetMinutes > 0 ? '-' : '+'}\${offsetHours
-			.toString()
-			.padStart(2, '0')}:\${offsetRestMinutes.toString().padStart(2, '0')}\`;
-	};
-
 	export const DateFromISODateStringIO = new Type<Date, string, unknown>(
 		'DateFromISODateString',
 		(u): u is Date => u instanceof Date,
 		(u, c) =>
 			either.chain(tstring.validate(u, c), s => {
-				const offset = new Date().getTimezoneOffset();
-				const d = new Date(\`\${s}T00:00:00\${getISOTimezoneOffsetString(offset)}\`);
+				const d = new Date(\`\${s}T00:00:00\`);
```

Comment:
@Lonelind This line affects your PR https://github.com/devexperts/swagger-codegen-ts/pull/100. Please check it.

Reply:
@Nedgeva the reason to add this function was the same: to get rid of the timezone dependency. To remove it completely seems wrong. Did you try to create something in one timezone before midnight and then switch and read the date after midnight from another timezone? There are a lot of paradoxes connected not only with daylight saving time shift but also around midnight in two adjacent timezones.

Reply:
@Lonelind I see two points here:
1) parsing date string to date object. Normally we shouldn't add any timezone so dates will be parsed properly to expected date in any timezone;
2) working with date objects across different timezones. This is application-specific scenario and I guess this is not what this utlity should be resonsible for.

> There are a lot of paradoxes connected not only with daylight saving time shift but also around midnight in two adjacent timezones.

Totally agree with you, but still i think `DateFromISODateStringIO` should be responsible only for proper date string parsing (and stringifying). If there wider range of problems that this utility should solve then it's probably better to write some tests / docs to make intentions clear.

Reply:
Yes, I was referring to tests at least describing those cases.
I made a little investigation: `new Date(year, month, day)` will create a local time date. ISO string is always UTC if it's not said otherwise which I was trying to do there, in the function 😁

Reply:
@Lonelind Wait, doesn't ISO date include timezone?

Reply:
@raveclassic if you don't specify it (ISO standard allows it), the `Date` object will treat timezone as `Z` i.e. UTC.

Reply:
So it's up to the end string value whether to specify the timezone or not, right? So we could just use https://github.com/gcanti/io-ts-types/blob/master/src/DateFromISOString.ts ?

Reply:
@raveclassic not exactly. This patch (made by me) originally was for `date` format that is just `YYYY-MM-DD` and it should stay the same wherever we are. `DateFromISOString` converts it to midnight UTC and then the browser treats it as a whole date. The problem is that JS doesn't have separate Date and DateTime type constructors working with DateTime by default if ISO string is passed in.

Reply:
Got it. Looks like we can't even use `new Date` constructor with string input according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#Timestamp_string
Even more, current implementation (as well as proposed in this PR) is incorrect since it just concats ISO8601 (already including time/timezone) with extra time/timezone etc.
We need to introduce consistent way to parse date according to https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#data-types which says that date-time should be in https://tools.ietf.org/html/rfc3339 (ISO8601)

Reply:
The date format from API should be in ISO8601 format, yes. There are separate `date` and `date-time` types specified in Swagger/OpenAPI. This transformation is for `date`, not `date-time`, so we expect to get `YYYY-MM-DD` format.

I still propose to use `new Date(year, month, day)` call as it is the one to create a date with local timezone by default.

Reply:
> This transformation is for date, not date-time

Oh, I didn't notice it, sorry.

@Nedgeva Looks like according to https://tools.ietf.org/html/rfc3339#section-5.6 `date` should indeed be parsed as `YYYY-MM-DD`.

@Lonelind Though I'm not sure whether we need to mess with timezones then. As ISO8601 (RFC) says timezone is only encoded by `date-time` where here we deal with just a `date`. It doesn't matter for the client in which timezone such value as received from the server. I'd even expect it to be UTC instead of local timezone, because, well, we have no idea of the client's timezone on backend. So if we'd need to pass correct _time_ to the client I'd mark the type as `date-time` and just use `parseISO`.

Reply:
@raveclassic yes, I've run a couple of tests in different timezones and `parseISO` seems to be what we're looking for.

The codec is created that way, that we send just a local date as well, without time. So yes, either `new Date(year, month, day)` or `parseISO` should work perfectly to parse the response. To form the request the codec uses `getYear`, `getMonth`, and `getDate` functions which return values in local time, so the conversion should work properly.

Reply:
I'd avoid extra dependencies such as `date-fns`, especially if parser is trivial - we just need `YYYY-MM-DD`

Reply:
Agree.

Reply:
@Nedgeva so, let's use `new Date(year, month, day)` constructor instead of ISO-string based one.